tests/test_main.py

```diff
@@ -236,6 +240,31 @@ def test_set_self_enrolment_method_key(moodle_requests_mock):
     assert result.exit_code == 0
 
 
+def test_export_grades(moodle_requests_mock, tmp_path, mocker):
+    runner = CliRunner()
+
+    s3_client = boto3.client('s3')
+    stubber = botocore.stub.Stubber(s3_client)
+
+    key = '/path/key.txt'
+    bucket_name = 'test-bucket'
+    data = {'courseid': 21, 'users': {'id': 3}}
+
+    expected_params = {'Bucket': bucket_name,
+                       'Body': json.dumps(data).encode('utf-8'),
+                       'Key': key}
+    stubber.add_response('put_object', {}, expected_params)
+    stubber.add_response('create_bucket', {}, expected_params)
+    stubber.activate()
+    mocker.patch("boto3.client", lambda service: s3_client)
+
+    result = runner.invoke(cli, ['export-grades',
+                                 '21', bucket_name, key],
+                           env=TEST_ENV)
+
+    assert result.exit_code == 0
```

Comment:
It might be good to add a `s3_stubber.assert_no_pending_responses()` here just as a check that anything expected to be called was.